src/main/java6/net/finmath/marketdata/model/curves/ForwardCurveFromDiscountCurve.java

```diff
@@ -18,119 +18,110 @@
-	 * The forward with fixing in t is calculated as ( df(t)/df(t+p)-1 ) / p
-	 * where df denotes the discount factor as a function of maturity and
-	 * p is a given the payment offset.
-	 * 
-	 * The date t + p is generated from the paymentOffsetCode using the provided paymentOffsetBusinessdayCalendar and paymentOffsetDateRollConvention.
+	 * Create a forward curve using a given referenceDiscountCurveForForwards.
 	 * 
 	 * @param name The name under which the forward curve can be referenced.
-	 * @param discountCurveName The discount curve used for calculation of the forward.
+	 * @param referenceDiscountCurveName The (pseudo-)discount curve that the forwards are calculated from.
+	 * @param discountCurveName The name of the discount curve associated with this forward curve (usually OIS).
 	 * @param referenceDate The reference date used in the interpretation of times (i.e., the referenceDate where t=0).
 	 * @param paymentOffsetCode The payment offset. If null, the parameter p has to be provided to the getForward method.
 	 * @param paymentOffsetBusinessdayCalendar The calendar used to generate the payment date from the paymentOffetCode.
 	 * @param paymentOffsetDateRollConvention The date roll convention used to generate the payment date from the paymentOffsetCode.
 	 * @param daycountScaling The scaling factor applied to the paymentOffset measured in ACT/365.
 	 * @param periodOffset An offset in ACT/365 applied to the fixing to construct the period start (the negative of the fixingOffset of the period).
 	 */
-	public ForwardCurveFromDiscountCurve(String name, String discountCurveName, LocalDate referenceDate, String paymentOffsetCode, BusinessdayCalendarInterface paymentOffsetBusinessdayCalendar, BusinessdayCalendarInterface.DateRollConvention paymentOffsetDateRollConvention, double daycountScaling, double periodOffset) {
+	public ForwardCurveFromDiscountCurve(String name, String referenceDiscountCurveName, String discountCurveName, LocalDate referenceDate, String paymentOffsetCode, BusinessdayCalendarInterface paymentOffsetBusinessdayCalendar, BusinessdayCalendarInterface.DateRollConvention paymentOffsetDateRollConvention, double daycountScaling, double periodOffset) {
 		super(name, referenceDate, paymentOffsetCode, paymentOffsetBusinessdayCalendar, paymentOffsetDateRollConvention, discountCurveName);
 
+		this.referenceDiscountCurveForForwardsName = referenceDiscountCurveName;
 		this.daycountScaling = daycountScaling;
 		this.periodOffset = periodOffset;
+		
+		System.out.println("ForwardCurveFromDiscountCurve(): Create forward curve " + name + " from referenceDiscountCurve " + referenceDiscountCurveName + " with discountCurveName=" + discountCurveName + ", referenceDate=" + referenceDate + ",  paymentOffsetCode=" + paymentOffsetCode + ", busDayCalendar=" + paymentOffsetBusinessdayCalendar + ", dateRollConvention=" + paymentOffsetDateRollConvention.name() + ", daycountScaling=" + daycountScaling + ", periodOffset=" + periodOffset); 

```

Comment:
Please do not add such printouts to the console to the code. It will pollute the console output. I did not recognize this change and was confused to see that unit test output have become corrupted.

System.out.println is OK in unit test or test programs, but not in a constructor of a frequently used class. For log messages use logger.